src/test/java/org/jenkinsci/plugins/workflow/cps/PersistenceProblemsTest.java

```diff
@@ -337,4 +338,25 @@ public void inProgressButStartBlocksLost() throws Exception {
             assertCompletedCleanly(run);
         });
     }
+
+    @Issue("JENKINS-50888")  // Tried to modify build without lazy load being triggered
+    @Test public void modifyBeforeLazyLoad() {
+        story.thenWithHardShutdown(r -> {  // Normal build
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("echo 'dosomething'", true));
+            r.buildAndAssertSuccess(p);
+        });
+        story.then(r -> {  // But wait, we try to modify the build without loading the execution
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            b.setDescription("Bob");
+            b.save();  // Will trigger an IOException potentially
```

Comment:
You mean, before the fix it might have thrown `IOException`? Comment is confusing in this context. If you are just reading the code, you would wonder—wait, if this throws an exception, would the test not fail now?